### PR TITLE
Feature/32 Projection link styling

### LINKED ISSUE #31: Projections styling
as per title, projections needs to be styled as per mockups.

As starting point please branch from feature/30.
The widget used for the projections is now specified in the [componentsMap](https://github.com/MetaCell/PsyNeuLinkView/blob/378216028af44971e043d4d66adf40d70c28c791/src/components/Main.js#L45), so feel free to apply the styling to the component [here](https://github.com/MetaCell/PsyNeuLinkView/blob/feature/30/src/components/custom-node/CustomLinkWidget.js)

### PR DESCRIPTION
**Pull request by vidhyalongani**
Closes #31
Screen Shot 2022-08-09 at 12 34 29 PM

### CODE REVIEW

File: src/components/views/projections/CustomLinkWidget.js [hunk 1]
```diff
@@ -1,63 +1,265 @@
...
+    generateLinePath(firstPoint, lastPoint) {
+        if (firstPoint.x <= lastPoint.x) {
+            return `M${firstPoint.x + 15},${firstPoint.y} L ${lastPoint.x},${lastPoint.y}`;
+        } else {
+            return `M${firstPoint.x - 15},${firstPoint.y} L ${lastPoint.x},${lastPoint.y}`;
+        }
     }
 
     render() {
+        //ensure id is present for all points on the path
```
Comment (zsinnema) [inline src/components/views/projections/CustomLinkWidget.js:87]: @vidhya-metacell can you explain what you are doing here and below? seems a little complex to me

**Review by zsinnema — CHANGES_REQUESTED** [review]

File: src/components/views/projections/CustomLinkWidget.js (hunk 1 above)
Comment (vidhyalongani) [reply src/components/views/projections/CustomLinkWidget.js:87]: @zsinnema , we are using our custom svg's for both projection link line and projection link arrow head to match with figma. So for calculating the position points to display these two @ddelpiano did all these calculations for points as well as the angle of the arrow head, in order to maintain the position of these links while we move the nodes on the canvas. May be @ddelpiano can explain better, if there any further questions on this.

Comment (zsinnema) [reply src/components/views/projections/CustomLinkWidget.js:87]: @ddelpiano I suggest to add comments to the calculations and move it to somewhere "generic"? Is this something generic special for PsyNeuLinkView or is it generic for MetaDiagram?

Comment (zsinnema) [reply src/components/views/projections/CustomLinkWidget.js:87]: @ddelpiano and also could you please remove the "fixed" constants like `pointlength` etc and replace it by either something dynamic calculated (depending on the source and target objects) or a configuration item

**Comment by zsinnema** [comment]
@vidhya-metacell can you try to resolve the merge conflicts? if not then please contact Dario, he did some changes that were merged before yours

**Comment by ddelpiano** [comment]
> @vidhya-metacell can you try to resolve the merge conflicts? if not then please contact Dario, he did some changes that were merged before yours

@zsinnema  we don't need this PR anymore, I brought all the changes that @vidhya-metacell did here in the other branch and PR already merged.

**Comment by zsinnema** [comment]
@ddelpiano thanks for that

**Pull request closed without merging**